### ISSUE TITLE
Question about `r` conditioning at inference

### ISSUE THREAD

**Issue by angusturner** [hugofloresgarcia/vampnet#13]
Hi, 

Just wondering whether the `r` value used to condition the model in the inference loop ought to depend on the initial masking fraction?

e.g) should we take `r = invgamma(initial_mask_fraction)`, as the first value of `r` in the iterator?  

https://github.com/hugofloresgarcia/vampnet/blob/a66dc9cb8aa8494f8d8ed53ac1e5bf99a6d6483e/vampnet/modules/transformer.py#L650C13-L650C13

**Comment by hugofloresgarcia** [comment]
Hi!

The dependency between `r` and the initial mask fraction is here: 
https://github.com/hugofloresgarcia/vampnet/blob/a66dc9cb8aa8494f8d8ed53ac1e5bf99a6d6483e/vampnet/modules/transformer.py#L702
```python
            num_to_mask = torch.floor(_gamma(r) * num_mask_tokens_at_start).unsqueeze(1).long()
```

`_gamma` is for going from a linear scale to cosine scale, and `invgamma` is for groing from a cosine scale to linear scale.  

inside the `generate` function, `r` is a number from 0-1 linearly scaled over the number of steps: 
https://github.com/hugofloresgarcia/vampnet/blob/a66dc9cb8aa8494f8d8ed53ac1e5bf99a6d6483e/vampnet/modules/transformer.py#L650
```
            # our current schedule step
            r = scalar_to_batch_tensor(
                (i + 1) / sampling_steps, 
                z.shape[0]
            ).to(z.device)
```


let me know if this clears things up!

**Comment by angusturner** [comment]
Hi, thanks for your reply.

My confusion is with an apparent discrepancy between training and generation with the `r` value. Specifically when you condition the model on this `r` value.

For example, at training `r` is randomly sampled between [0, 1] and corresponds with the fraction of masked tokens.

During inference, you always iterate `r` from 0->1, without regard for the initial masking fraction. For example, if we start the generation with 50% of tokens masked out, shouldn't we iterate from 0.5 -> 1 in that inference loop?

Hope this makes sense. :)

**Comment by angusturner** [comment]
Just to clarify further, it is this line that does not seem to have a dependence on the initial mask fraction: 

https://github.com/hugofloresgarcia/vampnet/blob/a66dc9cb8aa8494f8d8ed53ac1e5bf99a6d6483e/vampnet/modules/transformer.py#L663

**Comment by hugofloresgarcia** [comment]
Ah, I see! 

```
logits = self.forward(latents, r)
```

the r here doesn't actually do anything (if you look at `model.r_cond_dim`, you'll see that is set to 0 in our pretrained models, and it is also set to 0 in `conf/vampnet.yml`). 

This `r` term was here when we were experimenting with a diffusion-like architecture that needed the mask fraction to be passed as an argument. Sorry about having this outdated term in the forward pass, making things confusing!

I'm going to go ahead and remove that term from the forward pass, to make things clearer. 
should be now removed as of 41ea705db154e00743ca0a16aa4fe0d8116fc8e2

**Comment by hugofloresgarcia** [comment]
Eventually, I'd like to re-write the architecture to remove a bunch of components to the model from when we were doing diffusion vs maskgit exeperiments. I've made some progress on this branch: https://github.com/hugofloresgarcia/vampnet/tree/x-tfm, but haven't gotten to train a full model yet!

**Comment by angusturner** [comment]
Ah okay, cool! Thanks for clarifying that :)
